front/app/components/AssignBudgetControl/index.tsx

```diff
@@ -98,184 +87,181 @@ const StyledPBExpenses = styled(PBExpenses)`
+            participation_context_type: capitalizeParticipationContextType(
+              participationContextType
+            ),
+            idea_ids: newIdeas,
+            submitted_at: null,
+          });
+          done();
+          trackEventByName(tracks.ideaAddedToBasket);
+        } catch (error) {
+          done();
+          streams.fetchAllWith({ dataId: [basket.id] });
+        }
+      } else {
+        try {
+          await addBasket({
+            user_id: authUser.id,
+            participation_context_id: participationContextId,
+            participation_context_type: capitalizeParticipationContextType(
+              participationContextType
+            ),
+            idea_ids: [idea.id],
+          });
+          done();
+          trackEventByName(tracks.basketCreated);
+        } catch (error) {
+          done();
+        }
       }
     };
 
     if (
-      !isUndefined(authUser) &&
       !isNilOrError(idea) &&
       !isUndefined(basket) &&
-      idea.attributes.budget
+      idea.attributes.budget &&
+      participationContextId
     ) {
       const basketIdeaIds = !isNilOrError(basket)
         ? basket.relationships.ideas.data.map((idea) => idea.id)
         : [];
       const isInBasket = includes(basketIdeaIds, ideaId);
       const isBudgetingEnabled =
         idea.attributes.action_descriptor.budgeting?.enabled;
-      const isSignedIn = !!authUser;
-      const budgetingDisabledReason =
-        idea.attributes.action_descriptor.budgeting?.disabled_reason;
-      const isPermitted = budgetingDisabledReason !== 'not_permitted';
+      const isSignedIn = !isNilOrError(authUser);
+      const isPermitted =
+        idea.attributes.action_descriptor.budgeting?.disabled_reason !==
+        'not_permitted';
       const hasBudgetingDisabledReason =
         !!idea.attributes.action_descriptor.budgeting?.disabled_reason;
-      const isPBContext =
-        participationContext?.attributes?.participation_method === 'budgeting';
       const isCurrentPhase =
         getCurrentPhase(phases)?.id === participationContext?.id;
       const isCurrent =
         (participationContextType === 'project' &&
           (participationContext as IProjectData).attributes
             .publication_status !== 'archived') ||
         isCurrentPhase;
+      const buttonMessage = getAddRemoveButtonMessage(view, isInBasket);
 
-      const getAddRemoveButtonMessage = (view: TView) => {
-        switch (view) {
-          case 'ideaCard':
-            if (isInBasket) {
-              return messages.remove;
-            } else {
-              return messages.add;
-            }
-          case 'ideaPage':
-            if (isInBasket) {
-              return messages.removeFromMyBasket;
-            } else {
-              return messages.addToMyBasket;
-            }
-        }
-      };
-
-      if (isPBContext) {
+      if (
+        participationContext?.attributes?.participation_method === 'budgeting'
```

Comment:
Removed some unnecessary variables, where the path to the value was descriptive enough. I also moved the getAddRemoveButtonMessage outside of the component to clean it up some more.